src/lib/return-game-object-home.js

```diff
@@ -0,0 +1,48 @@
+const { ObjectNamespace } = require("./object-namespace");
+
+class ReturnGameObjectHome {
+  /**
+   * Static-only class, do not instantiate it.
+   */
+  constructor() {
+      throw new Error('Static only')
+  }
+
+  static returnAll(gameObjects) {
+    return gameObjects.forEach(gameObject => ReturnGameObjectHome.return(gameObject))
+  }
+
+  static return(gameObject) {
+    if(ObjectNamespace.isCard(gameObject)) {
+      return ReturnGameObjectHome.returnCard(gameObject)
+    }
+    if(ObjectNamespace.isCommandToken(gameObject)) {
+      return ReturnGameObjectHome.returnToPlayerReinforcements(gameObject)
+    }
+    if(ObjectNamespace.isControlToken(gameObject)) {
+      return ReturnGameObjectHome.returnToPlayerReinforcements(gameObject)
+    }
+    if(ObjectNamespace.isStrategyCard(gameObject)) {
+      return ReturnGameObjectHome.returnStrategyCard(gameObject)
+    }
+    if(ObjectNamespace.isSystemTile(gameObject)) {
+      return ReturnGameObjectHome.returnSystemTile(gameObject)
+    }
+    if(ObjectNamespace.isToken(gameObject)) {
+      return ReturnGameObjectHome.returnToken(gameObject)
+    }
+    if(ObjectNamespace.isUnit(gameObject)) {
+      return ReturnGameObjectHome.returnToPlayerReinforcements(gameObject)
+    }
+
+    throw new Error('Unable to return unknown object type')
```

Comment:
Toss in the gameObject.getTemplateMetadata() string in the error message, it'll cut later "what happened??" debugging.